src/NServiceBus.Core.Tests/Msmq/MsmqQueueCreatorTests.cs

```diff
@@ -0,0 +1,223 @@
+﻿namespace NServiceBus.Core.Tests.Msmq
+{
+    using System;
+    using System.Messaging;
+    using System.Security.Principal;
+    using NServiceBus.Core.Tests.Utils;
+    using NServiceBus.Transports.Msmq.Config;
+    using NUnit.Framework;
+    using Transports.Msmq;
+
+    [TestFixture]
+    public class MsmqQueueCreatorTests
+    {
+        [Test]
+        public void Should_create_all_queues()
+        {
+            var creator = GetCreator(true);
+
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.receiver");
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.endpoint1");
+            DeleteQueueIfPresent("MsmqQueueCreatorTests.endpoint2");
+
+            creator.CreateQueueIfNecessary(Address.Parse("MsmqQueueCreatorTests.receiver"), WindowsIdentity.GetCurrent().Name);
+            creator.CreateQueueIfNecessary(Address.Parse("MsmqQueueCreatorTests.target1"), WindowsIdentity.GetCurrent().Name);
+            creator.CreateQueueIfNecessary(Address.Parse("MsmqQueueCreatorTests.target2"), WindowsIdentity.GetCurrent().Name);
+
+            Assert.True(QueueExists("MsmqQueueCreatorTests.receiver"));
+            Assert.True(QueueExists("MsmqQueueCreatorTests.target1"));
+            Assert.True(QueueExists("MsmqQueueCreatorTests.target2"));
+        }
+
+        [Test]
+        public void Should_not_create_remote_queues()
```

Comment:
Talked to @andreasohlund and we found sources that claim we can never create remote private queues anyway which would mean this test is valid.